ludwig/models/trainer.py

```diff
@@ -91,183 +100,201 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         :type skip_save_model: Boolean
-        :param skip_save_progress: disables saving progress each epoch.
-               By default Ludwig saves weights and stats  after each epoch
-               for enabling resuming of training, but if the model is
-               really big that can be time consuming and will uses twice
-               as much space, use this parameter to skip it, but training
-               cannot be resumed later on
+        :param skip_save_progress: Disables saving progress each epoch. By default Ludwig saves weights and stats after
+               each epoch for enabling resuming of training, but if the model is really big that can be time consuming
+               and will uses twice as much space, use this parameter to skip it, but training cannot be resumed later
+               on. (default: False).
         :type skip_save_progress: Boolean
-        :param skip_save_log: Disables saving TensorBoard
-               logs. By default Ludwig saves logs for the TensorBoard, but if it
-               is not needed turning it off can slightly increase the
-               overall speed..
+        :param skip_save_log: Disables saving TensorBoard logs. By default Ludwig saves logs for the TensorBoard, but if
+               it is not needed turning it off can slightly increase the overall speed. (default: False).
         :type skip_save_log: Boolean
-        :param callbacks: a list of `ludwig.callbacks.Callback` objects that
-               provide hooks into the Ludwig pipeline.
-        :type: list
-        :param random_seed: Default initialization for the random seeds
-        :type: Float
-        :param horovod: Horovod parameters
+        :param callbacks: List of `ludwig.callbacks.Callback` objects that provide hooks into the Ludwig pipeline.
+               (default: None).
+        :type callbacks: list
+        :param random_seed: Default initialization for the random seeds (default: 42).
+        :type random_seed: Float
+        :param horovod: Horovod parameters (default: None).
         :type horovod: dict
-        :param device: The device to load the model on from a saved checkpoint.
+        :param device: Device to load the model on from a saved checkpoint (default: None).
         :type device: str
+        :param config: `ludwig.models.trainer.TrainerConfig` instance that specifies training hyperparameters (default:
+               `ludwig.models.trainer.TrainerConfig()`).
         """
-        self.epochs = epochs
-        self.regularization_lambda = regularization_lambda
-        self.regularization_type = regularization_type
-        self.learning_rate = learning_rate
+        if config is None:
+            config = TrainerConfig()
+
+        self.epochs = config.epochs
+        self.regularization_lambda = config.regularization_lambda
+        self.regularization_type = config.regularization_type
+        self.learning_rate = config.learning_rate
```

Comment:
@dantreiman Asking since you most recently touched `lr`, just wondering if appropriate for this logic to be simplified as part of this PR? In the context of the other changes, it is no longer the case that `learning_rate` can be non-`float`.